### PR TITLE
Add section about Smart Orchestration to the Best Practices page

### PR DESCRIPTION
**Pull request by debrisapron**
This addresses #4700.

Adds a Best Practices section suggesting use of the Dashboard to reduce test duration / load.

### CODE REVIEW

**Comment by debrisapron** [comment]
New para is here: https://cypress-documentation-224st5aeq-cypress-io.vercel.app/guides/references/best-practices#Running-Tests-Intelligently

File: content/guides/references/best-practices.md [hunk 1]
```diff
@@ -939,6 +939,24 @@ cy.wait('@getUsers') // <--- wait explicitly for this route to finish
 cy.get('table tr').should('have.length', 2)
 ```
 
+## Running Tests Intelligently
+
+As your test suite grows & takes longer to run, you may find yourself hitting
```
Comment (pstakoun) [inline content/guides/references/best-practices.md:944]:
```suggestion
As your test suite grows and takes longer to run, you may find yourself hitting
```

File: content/guides/references/best-practices.md [hunk 2]
```diff
@@ -939,6 +939,24 @@ cy.wait('@getUsers') // <--- wait explicitly for this route to finish
...
+performance bottlenecks on your CI system. We recommend integrating your source
+control system with your test suite such that merges are blocked until all your
+Cypress tests have passed, but the downside of this is that longer test
+execution times slow the velocity at which branches may be merged, and features
+may be shipped. This issue is compounded further if you have dependent chains of
```
Comment (pstakoun) [inline content/guides/references/best-practices.md:949]:
```suggestion
performance bottlenecks on your CI system. We recommend integrating your source
control system with your test suite such that merges are blocked until all your
Cypress tests have passed. The downside of this is that longer test
execution times slow the velocity at which branches may be merged and features
may be shipped. This issue is compounded further if you have dependent chains of
```

File: content/guides/references/best-practices.md [hunk 3]
```diff
@@ -939,6 +939,24 @@ cy.wait('@getUsers') // <--- wait explicitly for this route to finish
...
+branches waiting to be merged.
+
+One solution to this problem is
+[Smart Orchestration with the Cypress Dashboard](/guides/dashboard/smart-orchestration).
+Using a combination of [parallelization](/guides/guides/parallelization),
+[load balancing](/guides/guides/parallelization#Balance-strategy) &
```
Comment (pstakoun) [inline content/guides/references/best-practices.md:955]:
```suggestion
[load balancing](/guides/guides/parallelization#Balance-strategy), and
```

File: content/guides/references/best-practices.md [hunk 4]
```diff
@@ -939,6 +939,24 @@ cy.wait('@getUsers') // <--- wait explicitly for this route to finish
...
+[test run cancellation](/guides/dashboard/smart-orchestration#Cancel-test-run-when-a-test-fails),
```
Comment (pstakoun) [inline content/guides/references/best-practices.md:956]: Should we mention failure prioritization ([Run failed specs first](https://docs.cypress.io/guides/dashboard/smart-orchestration#Run-failed-specs-first)) here too?

File: content/guides/references/best-practices.md [hunk 5]
```diff
@@ -939,6 +939,26 @@ cy.wait('@getUsers') // <--- wait explicitly for this route to finish
...
+shipped. This issue is compounded further if you have dependent chains of
+branches waiting to be merged.
+
+One solution to this problem is
+[Smart Orchestration with the Cypress Dashboard](/guides/dashboard/smart-orchestration).
+Using a combination of [parallelization](/guides/guides/parallelization),
+[load balancing](/guides/guides/parallelization#Balance-strategy), and
+[test run cancellation](/guides/dashboard/smart-orchestration#Cancel-test-run-when-a-test-fails),
+and
+[running failed specs first](/guides/dashboard/smart-orchestration#Run-failed-specs-first),
```
Comment (pstakoun) [inline content/guides/references/best-practices.md:958]: there are two "and"s here

**Review by pstakoun — CHANGES_REQUESTED** [review]
just that double "and"

**Review by pstakoun — APPROVED** [review]

**Comment by pstakoun** [comment]
I think it would be cool to use one of the smart orchestration/parallelization images/diagrams here to call more attention to the section, but we can improve this iteratively. lgtm

**Pull request merged**